deps/picotls/lib/openssl.c

```diff
@@ -693,77 +695,138 @@ int ptls_openssl_create_key_exchange(ptls_key_exchange_context_t **ctx, EVP_PKEY
     }
 }
 
+struct sign_ctx {
+    const struct st_ptls_openssl_signature_scheme_t *scheme;
+    EVP_MD_CTX *ctx;
+    ptls_buffer_t buf;
+    size_t siglen;
+    ASYNC_WAIT_CTX *waitctx;
+    ASYNC_JOB *job;
+};
+
+int ptls_openssl_get_async_fd(ptls_t *ptls)
+{
+    int fds[1];
+    size_t numfds;
+    struct sign_ctx *args = ptls_get_sign_ctx(ptls);
+    ASYNC_WAIT_CTX_get_all_fds(args->waitctx, NULL, &numfds);
+    assert(numfds == 1);
+    ASYNC_WAIT_CTX_get_all_fds(args->waitctx, fds, &numfds);
+    return fds[0];
+}
```

Comment:
If we could change the design so that the application would inject `ASYNC_WAIT_CTX`, I think we can get rid of his function?

Reply:
https://github.com/h2o/h2o/pull/3064#discussion_r927895067